index.ts

```diff
@@ -161,49 +156,51 @@ class Example {
 class ExampleWithOptions {
 	index = 0
 
-	@mem.decorator({maxAge: 1000})
+	@memDecorator({maxAge: 1000})
 	counter() {
 		return ++this.index;
 	}
 }
 ```
 */
-mem.decorator = <
+export function memDecorator<
 	FunctionToMemoize extends AnyFunction,
-	CacheKeyType
+	CacheKeyType,
 >(
-	options: Options<FunctionToMemoize, CacheKeyType> = {}
-) => (
-	target: any,
-	propertyKey: string,
-	descriptor: PropertyDescriptor
-): void => {
-	const input = target[propertyKey];
-
-	if (typeof input !== 'function') {
-		throw new TypeError('The decorated value must be a function');
-	}
+	options: Options<FunctionToMemoize, CacheKeyType> = {},
+) {
+	return (
+		target: any,
+		propertyKey: string,
+		descriptor: PropertyDescriptor,
+	): void => {
+		const input = target[propertyKey]; // eslint-disable-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
```

Comment:
These errors must be ignored because `target: any` and there is currently no viable alternative.
https://stackoverflow.com/questions/68343539/typescript-definition-of-target-in-a-decorator